runtime/unix.js

```diff
@@ -64,3 +64,180 @@ function unix_isatty(fileDescriptor) {
     return false;
   }
 }
+
+//Provides: caml_raise_unix_exn_of_nodejs_error
+//Requires: caml_raise_with_args
+//Requires: caml_named_value, caml_string_of_jsbytes
+function caml_raise_unix_exn_of_nodejs_error(err) {
+  /* ===Unix.error===
+   *
+   * This array is in order of the variant in OCaml
+   */
+  var errors = [
+    'E2BIG', 'EACCES', 'EAGAIN', 'EBADF', 'EBUSY', 'ECHILD', 'EDEADLK', 'EDOM',
+    'EEXIST', 'EFAULT', 'EFBIG', 'EINTR', 'EINVAL', 'EIO', 'EISDIR', 'EMFILE',
+    'EMLINK', 'ENAMETOOLONG', 'ENFILE', 'ENODEV', 'ENOENT', 'ENOEXEC', 'ENOLCK',
+    'ENOMEM', 'ENOSPC', 'ENOSYS', 'ENOTDIR', 'ENOTEMPTY', 'ENOTTY', 'ENXIO',
+    'EPERM', 'EPIPE', 'ERANGE', 'EROFS', 'ESPIPE', 'ESRCH', 'EXDEV', 'EWOULDBLOCK',
+    'EINPROGRESS', 'EALREADY', 'ENOTSOCK', 'EDESTADDRREQ', 'EMSGSIZE',
+    'EPROTOTYPE', 'ENOPROTOOPT', 'EPROTONOSUPPORT', 'ESOCKTNOSUPPORT',
+    'EOPNOTSUPP', 'EPFNOSUPPORT', 'EAFNOSUPPORT', 'EADDRINUSE', 'EADDRNOTAVAIL',
+    'ENETDOWN', 'ENETUNREACH', 'ENETRESET', 'ECONNABORTED', 'ECONNRESET', 'ENOBUFS',
+    'EISCONN', 'ENOTCONN', 'ESHUTDOWN', 'ETOOMANYREFS', 'ETIMEDOUT', 'ECONNREFUSED',
+    'EHOSTDOWN', 'EHOSTUNREACH', 'ELOOP', 'EOVERFLOW'
+  ];
+  var variant = errors.indexOf(err.code);
+  if (variant < 0) {
+    // TODO: Make sure this works `EUNKNOWNERR(int)`
```

Comment:
This comment needs to be addressed

Reply:
This was an old comment, I'm 99% sure this is how the variant would be constructed. I don't know how to actually force this type of error in nodejs, and I don't think we can mock it. I removed the comment in my refactor.